rules_python_generator/wheel_generator.py

```diff
@@ -0,0 +1,479 @@
+                link = match.group('link')
+                filename = self._get_wheelname_from_link(link)
+                logger.debug('Found link: %s for: %s', link, filename)
+                wheel_links[filename] = link
+        end = time.time()
+        logger.info('pip executed in %s', (end - start) * 1000.0)
+        logger.debug('found: %r', wheel_links)
+        return wheel_links
+
+    def _parse_wheel_dependencies(self, wheel_dir: str,
+                                  wheel_links: str) -> typing.Set[DependencyInfo]:
+
+        # used to lookup links for wheels that also exist on PyPI, note that
+        # this could potentially include versions and platforms we don't care
+        # about
+        all_deps = set([])  # type: DependencyInfo
+
+        # at this point pip has installed all deps+subdeps inside our
+        # wheel_dir, even though we're dealing with wheel files, we are
+        # really iterating through our full dependency set
+        for wheel_filepath in glob.glob('{}/*.whl'.format(wheel_dir)):
+            logger.info("\nProcessing wheelinfo for %s", wheel_filepath)
+            wheel = Wheel(wheel_filepath)
+            extra_deps = {}
+            for extra in wheel.extras():
+                extra_deps[extra] = list(wheel.dependencies(extra=extra))
+
+            logger.debug("Wheel name is: %s", wheel.name())
+            dependency = DependencyInfo(
+                name=wheel.name(), deps=set(wheel.dependencies()) - BLACKLIST, extras=extra_deps
+            )
+
+            wheel_filename = os.path.basename(wheel_filepath)
+
+            # now we're going to try to find additional wheels for the same
+            # library+version on other platforms based on the wheel links
+            # that we got from the pip call
+            match = WHEEL_FILE_RE.search(wheel_filename)
+            if match:
+                name, version = match.group('name'), match.group('ver')
+                match_prefix = '{}-{}'.format(name, version)
+            else:
+                logger.error("Could not parse wheel file name.")
+                sys.exit(1)
+            logger.debug(
+                "Will find additional wheels for other platforms using prefix: %s", match_prefix
+            )
+            for additional_filename, additional_link in wheel_links.items():
+                if additional_filename.startswith(match_prefix):
+                    # we've found a wheel with the same name+version, obviously
+                    # it could just be the wheel we downloaded (the local wheel)
+                    # so we check for that first.
+                    logger.info("Found additional wheel: %s", additional_filename)
+
+                    is_compatible = check_compatibility(additional_filename)
+                    if not is_compatible:
+                        continue
+
+                    filepath = os.path.abspath(self.wheel_dir + '/' + additional_filename)
```

Comment:
`wheel_dir` instead of `self.wheel_dir` (unless you get rid of `wheel_dir` as an argument). Also maybe use `os.path.join` instead of `+`